helloworld.c

```diff
@@ -0,0 +1,83 @@
+#include <malamute.h>
+
+static char *endpoint = "inproc://bob";
+
+void
+provider_server (zsock_t *pipe, void* args) {
+
+    char *name = strdup (args);
+
+    int r;
+    mlm_client_t *provider = mlm_client_new ();
+    r = mlm_client_connect (provider, endpoint, 1000, name);
+    assert (r != -1);
+
+    zsock_signal (pipe, 0);
+    while (!zsys_interrupted) {
+        zmsg_t *msg = mlm_client_recv (provider);
+        zsys_info ("Message received:");
```

Comment:
You should create zpoller_t and zpoller_wait method. Actor must be able to react on pipe.